pulp_container/app/authorization.py

```diff
@@ -132,29 +136,21 @@ def determine_access(self):
         typ, name, actions = self.scope.split(":")
         actions = set(actions.split(","))
 
-        permitted_actions = set()
-        for action in actions:
-            has_permission = self.actions_permissions[action](name)
-            if has_permission:
-                permitted_actions.add(action)
+        permitted_actions = {action for action in actions if self.actions_permissions[action](name)}
 
         return {"type": typ, "name": name, "actions": list(permitted_actions)}
 
     def has_pull_permissions(self, path):
         """
         Check if the user has permissions to pull from the repository specified by the path.
         """
-        if self.user.has_perm("container.view_containerdistribution"):
-            return True
-
-        try:
-            distribution = ContainerDistribution.objects.get(base_path=path)
-        except ContainerDistribution.DoesNotExist:
-            return False
-        if self.user.has_perm("container.view_containerdistribution", distribution):
-            return True
-
-        return False
+        # Fake the pull request
+        request = Request(HttpRequest())
+        request.method = "GET"
+        request.user = self.user
+        # TODO Fake the corresponding view
+        view = APIView()
+        return self.access_policy.has_permission(request, view)
```

Comment:
I would keep going with this for now until we have a better solution. I've been working on extending rbac to not require NamedModelViewSet but it's not ready just yet. The issue I'm working on is: https://pulp.plan.io/issues/8048

Reply:
It doesn't matter here, because we need the permissions for the registry endpoints in the token granting endpoint (totally weird).